In [1]:
import pandas as pd

df = pd.read_csv("../data/raw/Online-eCommerce.csv")
df.head()


Order_Number State_Code   Customer_Name  Order_Date     Status  \
0      139374.0         AP     Adhir Samal  11/01/2020  Delivered   
1      139375.0         AP  Dannana Jhammi  11/01/2020  Delivered   
2      139376.0         AS     Vipin Kumar  11/01/2020  Delivered   
3      139377.0         BR   Ranjeet Kumar  11/01/2020  Delivered   
4      139378.0         CG   Sajal Singhal  11/01/2020      Order   

                    Product      Category     Brand    Cost    Sales  \
0                512 GB M.2           SSD   Samsung  6500.0   8450.0   
1      RYZEN 3rd gen. 3500            CPU     Intel  8500.0  11050.0   
2          2GB Graphic Card  Graphic Card    Nvidia  7000.0   9100.0   
3            16 GB DDR4 RAM           RAM     Hynix  6550.0   8515.0   
4  Standard ATX motherboard   MotherBoard  Gigabyte  7650.0   9945.0   

   Quantity  Total_Cost  Total_Sales Assigned Supervisor  
0       1.0      6500.0       8450.0         Ajay Sharma  
1       3.0     25500.0      33150.0        Roshan Kumar  
2       2.0     14000.0      18200.0         Aarvi Gupta  
3       3.0     19650.0      25545.0         Vijay Singh  
4       4.0     30600.0      39780.0          Aadil Khan

In [2]:
df.shape

(5110, 14)

In [3]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 14 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Order_Number         5095 non-null   float64
 1   State_Code           5095 non-null   str    
 2   Customer_Name        5095 non-null   str    
 3   Order_Date           5095 non-null   str    
 4   Status               5095 non-null   str    
 5   Product              5095 non-null   str    
 6   Category             5095 non-null   str    
 7   Brand                5095 non-null   str    
 8   Cost                 5095 non-null   float64
 9   Sales                5095 non-null   float64
 10  Quantity             5095 non-null   float64
 11  Total_Cost           5095 non-null   float64
 12  Total_Sales          5095 non-null   float64
 13  Assigned Supervisor  5095 non-null   str    
dtypes: float64(6), str(8)
memory usage: 559.0 KB


In [4]:
df.columns

Index(['Order_Number', 'State_Code', 'Customer_Name', 'Order_Date', 'Status',
       'Product', 'Category', 'Brand', 'Cost', 'Sales', 'Quantity',
       'Total_Cost', 'Total_Sales', 'Assigned Supervisor'],
      dtype='str')

In [5]:
df["Status"].value_counts()

Status
Processing    1278
Order         1276
Shipped       1273
Delivered     1268
Name: count, dtype: int64

In [6]:
(df.isna().sum()
   .sort_values(ascending=False)
   .head(20))

Order_Number           15
State_Code             15
Customer_Name          15
Order_Date             15
Status                 15
Product                15
Category               15
Brand                  15
Cost                   15
Sales                  15
Quantity               15
Total_Cost             15
Total_Sales            15
Assigned Supervisor    15
dtype: int64

In [7]:
df["Order_Number"].nunique(), len(df)

(5095, 5110)

In [8]:
df["Order_Number"].value_counts().head(10)

Order_Number
139374.0    1
139375.0    1
139376.0    1
139377.0    1
139378.0    1
139379.0    1
139380.0    1
139381.0    1
139382.0    1
139383.0    1
Name: count, dtype: int64

In [9]:
dates = pd.to_datetime(df["Order_Date"], errors="coerce", dayfirst=True)
dates.min(), dates.max(), dates.isna().sum()

(Timestamp('2020-01-11 00:00:00'),
 Timestamp('2022-12-31 00:00:00'),
 np.int64(15))

In [10]:
(df["Sales"] * df["Quantity"]).head(), df["Total_Sales"].head()

(0     8450.0
 1    33150.0
 2    18200.0
 3    25545.0
 4    39780.0
 dtype: float64,
 0     8450.0
 1    33150.0
 2    18200.0
 3    25545.0
 4    39780.0
 Name: Total_Sales, dtype: float64)

In [11]:
df["Status"].value_counts(normalize=True)

Status
Processing    0.250834
Order         0.250442
Shipped       0.249853
Delivered     0.248871
Name: proportion, dtype: float64

Initial observations

- Dataset has 5110 rows and 14 columns.
- There are 5095 unique orders → some orders contain multiple items.
- About 15 rows contain missing values across all columns (likely empty rows).
- Order dates span from 2020 to 2022.
- Status values: Processing, Order, Shipped, Delivered.
- Data is suitable for e-commerce analytics and will be cleaned in the next step.
